token-swap/program/src/processor.rs

```diff
@@ -345,6 +391,7 @@ impl Processor {
             return Err(ProgramError::IncorrectProgramId);
         }
         let token_swap = SwapVersion::unpack(&swap_info.data.borrow())?;
+        Self::check_allowed_to_use(token_swap.as_ref(), 0)?;
```

Comment:
take it or leave it, but a better alternative to hard-coding these bit numbers would be to use a true bitmask library.  bitflags seems to be a pretty good one https://docs.rs/bitflags/1.2.1/bitflags/